src/pages/library/datasetBuilder/CohortEditor.ts

```diff
@@ -0,0 +1,429 @@
+import _ from 'lodash/fp';
+import { Fragment, useState } from 'react';
+import { div, h, h2, h3, strong } from 'react-hyperscript-helpers';
+import { ButtonOutline, ButtonPrimary, Link, Select } from 'src/components/common';
+import { icon } from 'src/components/icons';
+import { DatasetResponse } from 'src/libs/ajax/DatasetBuilder';
+import colors from 'src/libs/colors';
+import * as Utils from 'src/libs/utils';
+import {
+  Cohort,
+  Criteria,
+  CriteriaGroup,
+  DatasetBuilderState,
+  DomainCriteria,
+  DomainType,
+  newCriteriaGroup,
+  ProgramDataListCriteria,
+  ProgramDataListType,
+  ProgramDataRangeCriteria,
+  ProgramDataRangeType,
+} from 'src/pages/library/datasetBuilder/dataset-builder-types';
+import { HomepageState, OnStateChangeType } from 'src/pages/library/datasetBuilder/DatasetBuilder';
+import { datasetBuilderCohorts } from 'src/pages/library/datasetBuilder/state';
+
+const PAGE_PADDING_HEIGHT = 0;
+const PAGE_PADDING_WIDTH = 3;
+
+const renderCriteria = (deleteCriteria: (criteria) => void) => (criteria: Criteria) =>
```

Comment:
Did you think about doing this as a component? E.g.,
```
const Criteria = ({ deleteCriteria, criteria }: { deleteCriteria: (criteria: Criteria) => void, criteria: Criteria }) => return div(...)
```
I think that might make the map that uses it a little clearer.

Reply:
I agree as a general rule... lots of example of "render helpers" that would be better written as a standard component.

Reply:
Addressed in latest commits